File: problems/submatrix/scaffold.cpp
#include <iostream>
#include <vector>
using namespace std;

pair<pair<int, int>, int> findLargestSquareSubMatrix(vector<vector<int>>& matrix) {
    // Your solution goes here
    return make_pair(make_pair(x, y), max_size);
}

vector<vector<int>> parseMatrix() {
    int size;
    cin >> size;
    vector<vector<int>> matrix(size, vector<int>(size));

    for (int i = 0; i < size; i++) {
        for (int j = 0; j < size; j++) {
            cin >> matrix[i][j];
        }
    }

    return matrix;
}

int main() {
    vector<vector<int>> matrix = parseMatrix();
    pair<pair<int, int>, int> answer = findLargestSquareSubMatrix(matrix);
    cout << answer.second << endl;
    cout << answer.first.first << " " << answer.first.second << endl;
    return 0;
}

File: problems/6-submatrix/scaffold.py
def find_largest_square_submatrix(matrix):
    # Your code goes here
    return ((x, y), max_size)

def parse_matrix():
    size = int(input())
    matrix = []

    for _ in range(size):
        line = input()
        if not line:  # Stop reading if an empty line is encountered
            break

        matrix.append([int(x) for x in line.split(" ")])

    return matrix

answer = find_largest_square_submatrix(parse_matrix())
print(answer[1])
print(str(answer[0][0]) + " " + str(answer[0][1]))